src/groupederror.jl

```diff
@@ -1,15 +1,14 @@
 ##### List of functions to analyze data
 
 function extend_axis(df::AbstractDataFrame, xlabel, ylabel, xaxis, val)
-    aux = DataFrame()
-    aux[xlabel] = xaxis
+    aux = DataFrame(Any[xaxis],[xlabel])
     extended = join(aux, df, on = xlabel, kind = :left)
     sort!(extended, cols = [xlabel])
     return convert(Array, extended[ylabel], val)
 end
 
 function extend_axis(xsmall, ysmall, xaxis, val)
-    df = DataFrame(x = xsmall, y = ysmall)
+    df = DataFrame(Any[xsmall,ysmall], [:x,:y])
```

Comment:
I think I do: if I don't use it, it may happen that I try and join a dataframe of nullablearrays with a dataframe of normal arrays, which gives an error. The approach of `groupapply` is to convert what the user gave to a df of only arrays and then only work with that type of dataframes.